src/daemon/Display.cpp

```diff
@@ -421,7 +435,9 @@ namespace SDDM {
         m_sessionTerminalId = m_terminalId;
         if ((session.type() == Session::WaylandSession && m_displayServerType == X11DisplayServerType) || (m_greeter->isRunning() && m_displayServerType != X11DisplayServerType)) {
             // Create a new VT when we need to have another compositor running
-            m_sessionTerminalId = VirtualTerminal::setUpNewVt();
+            if (seat()->canTty()) {
+                m_sessionTerminalId = VirtualTerminal::setUpNewVt();
```

Comment:
The `env.insert(QStringLiteral("XDG_VTNR"), QString::number(m_sessionTerminalId));` below should probably be conditional depending on `m_sessionTerminalId > 0`

Reply:
To clarify: Currently it ends up with `XDG_VTNR=-1` which causes the helper to open `/dev/tty-1` and then go into the tty-less code path. That does kind of work but is not really how it should be...

Reply:
done locally